lib/handlers/PatchHandler.js

```diff
@@ -41,7 +41,8 @@ class PatchHandler extends BaseHandler {
         }
 
         const new_offset = await this.store.write(req, file_id, offset);
-        if (`${new_offset}` === stats.upload_length) {
+        const upload_length = stats.upload_length || req.headers['upload-length'];
+        if (parseInt(new_offset, 10) === parseInt(upload_length, 10)) {
```

Comment:
This approach would not work properly for S3 which need this information to split parts to chunks and only last chunk can be <5MB. If S3 can not determine that this is in fact last part it will be discarded and write will not finish.

Upload length should be written to datastore's configstore before any write is performed. Once this header is present it MUST be present and unchanged (and validated by the server) in all subsequent request.

From TUS protocol:
> If the length was deferred using Upload-Defer-Length: 1, the Client MUST set the Upload-Length header in the next PATCH request, once the length is known. Once set the length MUST NOT be changed. As long as the length of the upload is not known, the Server MUST set Upload-Defer-Length: 1 in all responses to HEAD requests